src/tools/tidy/src/alphabetical.rs

```diff
@@ -143,3 +145,70 @@ pub fn check(path: &Path, bad: &mut bool) {
         check_lines(file, lines, &mut crate::tidy_error, bad)
     });
 }
+
+fn consume_numeric_prefix<I: Iterator<Item = char>>(it: &mut Peekable<I>) -> String {
+    let mut result = String::new();
+
+    while let Some(&c) = it.peek() {
+        if c.is_numeric() {
+            result.push(c);
+            it.next();
+            continue;
+        }
+
+        break;
```

Comment:
```suggestion
        } else {
            break;
        }
```

or swap the `if` to `if !c.is_numeric() { break; }` ?